c2rust-analyze/src/context.rs

```diff
@@ -357,7 +361,16 @@ impl<'a, 'tcx> AnalysisCtxt<'a, 'tcx> {
     }
 
     pub fn project(&self, lty: LTy<'tcx>, proj: &PlaceElem<'tcx>) -> LTy<'tcx> {
-        util::lty_project(lty, proj)
+        let adt_func = |adtdef: AdtDef, field: Field| {
+            let fielddef = adtdef.all_fields().nth(usize::from(field)).unwrap();
+            let fielddef_name = fielddef.name;
+            eprintln!("projecting into {adtdef:?}.{fielddef_name:}");
+            let res = *self.gacx.field_tys.get(&fielddef.did).unwrap_or_else(|| {
+                panic!("Could not find {adtdef:?}.{fielddef_name:?} in field type map")
+            });
+            res
+        };
```

Comment:
```suggestion
        let adt_func = |adt_def: AdtDef, field: Field| {
            let field_def = adt_def.all_fields().nth(usize::from(field)).unwrap();
            let field_def_name = field_def.name;
            eprintln!("projecting into {adt_def:?}.{field_def_name:}");
            let res = *self.gacx.field_tys.get(&field_def.did).unwrap_or_else(|| {
                panic!("Could not find {adt_def:?}.{field_def_name:?} in field type map")
            });
            res
        };
```

See comment in https://github.com/immunant/c2rust/pull/726#pullrequestreview-1182354527.